pkg/logs/client/http/destination.go

```diff
@@ -393,12 +395,34 @@ func (d *Destination) updateRetryState(err error, isRetrying chan bool) bool {
 }
 
 func httpClientFactory(timeout time.Duration, cfg pkgconfigmodel.Reader) func() *http.Client {
+
+	transport := httputils.CreateHTTPTransport(cfg)
+	transportConfig := cfg.Get("logs_config.transport_type")
+	// Configure transport based on user setting
+	switch transportConfig {
+	case "http1":
+		// Use default ALPN auto-negotiation to negotiate up to http/1.1
+	case "auto":
+		fallthrough
+	default:
+		if cfg.Get("logs_config.transport_type") != "auto" {
+			log.Warnf("Invalid transport_type '%v', falling back to 'auto'", transportConfig)
+		}
+		// Use default ALPN auto-negotiation and negotiate to HTTP/2 if possible, if not it will automatically fallback to best available protocol
+		err := http2.ConfigureTransport(transport)
+		if err != nil {
+			log.Warnf("Failed to configure HTTP/2 transport: %v. Resolving to best available protocol", err)
+		}
+	}
```

Comment:
Do we see this code being used onother parts of the codebase? 

Currently, logs is the only one using it, but could there be a case that other client wants to enforce the protocol?

What about changing the `httputils.CreateHTTPTransport` function to accept an option, so we can tweak the http protocol. 

`httputils.CreateHTTPTransport(cfg, httputils.WithHttp2())`

by using the options patterrn we do not have to modify the other callers, and we can leave the http specifc code inside the `httputils ` package. 

The resulting code could look like this: 

```go
var transport *http.Transport

transportConfig := cfg.Get("logs_config.transport_type")
// Configure transport based on user setting
switch transportConfig {
case "http1":
	// Use default ALPN auto-negotiation to negotiate up to http/1.1
	transport = httputils.CreateHTTPTransport(cfg)
case "auto":
	fallthrough
default:
	if cfg.Get("logs_config.transport_type") != "auto" {
		log.Warnf("Invalid transport_type '%v', falling back to 'auto'", transportConfig)
	}
	// Use default ALPN auto-negotiation and negotiate to HTTP/2 if possible, if not it will automatically fallback to best available protocol
	transport = httputils.CreateHTTPTransport(cfg, httputils.WithHTTP2())
}
```

That way others can force HTTP2 as well and we do not have to duplicate this logic somewhere else